File: src/test/java/leetcode_challenges/april/day28/LCFirstUniqueTest.java
package leetcode_challenges.april.day28;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class LCFirstUniqueTest {

    private LCFirstUnique firstUnique;
    public LCFirstUniqueTest() {
        this.firstUnique = new LCFirstUnique(new int[]{2,3,5});
    }

    @Test
    public void testFirstUnique() {
        assertEquals(firstUnique.showFirstUnique(), 2);
        firstUnique.add(5);
        assertEquals(firstUnique.showFirstUnique(), 2);
        firstUnique.add(2);
        assertEquals(firstUnique.showFirstUnique(), 3);
        firstUnique.add(3);
        assertEquals(firstUnique.showFirstUnique(), -1);
    }
}

File: src/test/scala/leetcode_challenges/april/day28/LC_FirstUniqueTest.scala
package leetcode_challenges.april.day28

import org.scalatest.FlatSpec

class LC_FirstUniqueTest extends FlatSpec{

  "LC_FirstUnique" should "add and show unique values properly." in {
    val firstUnique = new LC_FirstUnique(Array(2, 3, 5))
    assert(firstUnique.showFirstUnique === 2)
    firstUnique.add(5)
    assert(firstUnique.showFirstUnique === 2)
    firstUnique.add(2)
    assert(firstUnique.showFirstUnique === 3)
    firstUnique.add(3)
    assert(firstUnique.showFirstUnique === -1)
  }
}
